.githooks/pre-push

```diff
@@ -2,16 +2,25 @@
 
 set -eu
 
-# run linter
-
 if ! [ -x "$(command -v npm)" ]; then
   echo 'Error: npm not found. Make it available to the host shell (e.g., with "nvm use --lts").'
   exit 2
 fi
 
+
+reset() {
+  echo "Restoring working tree"
+  git stash pop
+}
+
+if [ -n "$(git status --porcelain)" ]; then
+  echo "Working tree is not clean, temporary reset"
+  git stash save --keep-index --include-untracked
+  trap reset EXIT
+fi
```

Comment:
```suggestion
if [ -n "$(git status --porcelain)" ]; then
  echo "Working tree is not clean, make a temporary copy"
  tmpdir=$(mktemp --directory)
  cp -r . $tmpdir
  cd $tmpdir
  git stash save --quiet --keep-index --include-untracked
fi
```

Reply:
My version has a speedier copy by obeying the gitignore (the following reset is redundant as well but i placed it there to be save). Furthermore, i clean up with the reset trap to avoid overfilling the tmp folder.

Reply:
> to avoid overfilling the tmp folder

Looks like you've had bad experiences with /tmp getting cluttered xD (cough #395)

Reply:
*cries in R*